### PR TITLE
[flutter_adaptive_scaffold] exchange BottomNavigationBar with NavigationBar

### PR DESCRIPTION
**Pull request by fmt-Println-MKO**
To have a common Material 3 Design, BottomNavigationBar was replaced with M3 NavigationBar as adaptive scaffold claims to be Material 3.

[fixes: #124984](https://github.com/flutter/flutter/issues/124984)

## Pre-launch Checklist
* [x]  I read the [Contributor Guide](https://github.com/flutter/packages/blob/main/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
* [x]  I read the [Tree Hygiene](https://github.com/flutter/flutter/wiki/Tree-hygiene) wiki page, which explains my responsibilities.
* [x]  I read and followed the [relevant style guides](https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style) and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
* [x]  I signed the [CLA](https://cla.developers.google.com/).
* [x]  The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
* [x]  I listed at least one issue that this PR fixes in the description above.
* [x]  I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning), or this PR is [exempt from version changes](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates).
* [x]  I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style).
* [x]  I updated/added relevant documentation (doc comments with `///`).
* [x]  I added new tests to check the change I am making, or this PR is [test-exempt](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests).
* [x]  All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter_adaptive_scaffold/pubspec.yaml [hunk 1]
```diff
@@ -14,4 +14,4 @@ dependencies:
 
 dev_dependencies:
   flutter_test:
-    sdk: flutter
+    sdk: flutter
```
Comment (gspencergoog) [inline packages/flutter_adaptive_scaffold/pubspec.yaml:17]: Missing a newline at the end of the file.

File: packages/flutter_adaptive_scaffold/CHANGELOG.md [hunk 2]
```diff
@@ -58,4 +62,4 @@ Make fuchsia a mobile platform.
 
 ## 0.0.1
 
-* Initial release
+* Initial release
```
Comment (gspencergoog) [inline packages/flutter_adaptive_scaffold/CHANGELOG.md:65]: What is the diff on this line? Is there a missing newline?

File: packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart [hunk 3]
```diff
@@ -317,13 +317,10 @@ class AdaptiveScaffold extends StatefulWidget {
   }) {
     return Builder(
       builder: (_) {
-        return BottomNavigationBar(
-          currentIndex: currentIndex ?? 0,
-          iconSize: iconSize,
```
Comment (gspencergoog) [inline packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart:322]: It looks like the updated code ignores `iconSize` now. Could you reintroduce it with the `NavigationBarThemeData.iconTheme`?

File: packages/flutter_adaptive_scaffold/pubspec.yaml (hunk 1 above)
Comment (fmt-Println-MKO) [reply packages/flutter_adaptive_scaffold/pubspec.yaml]: thx, fixed

File: packages/flutter_adaptive_scaffold/CHANGELOG.md (hunk 2 above)
Comment (fmt-Println-MKO) [reply packages/flutter_adaptive_scaffold/CHANGELOG.md]: yes, thx, fixed

File: packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart (hunk 3 above)
Comment (fmt-Println-MKO) [reply packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart:322]: added

File: packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart [hunk 4]
```diff
@@ -317,14 +317,18 @@ class AdaptiveScaffold extends StatefulWidget {
...
-          items: destinations
-              .map((NavigationDestination e) => _toBottomNavItem(e))
-              .toList(),
-          onTap: onDestinationSelected,
-        );
+        return NavigationBarTheme(
+            data: NavigationBarThemeData(
+              iconTheme: MaterialStateProperty.resolveWith(
+                  (Set<MaterialState> states) {
+                return IconThemeData(size: iconSize);
```
[the remark this answers is not in the document]
Comment (fmt-Println-MKO) [reply packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart]: thx, I fixed it

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart [hunk 5]
```diff
@@ -316,15 +316,22 @@ class AdaptiveScaffold extends StatefulWidget {
...
+              return currentNavBarTheme.iconTheme
+                      ?.resolve(states)
+                      ?.copyWith(size: iconSize) ??
+                  IconTheme.of(context).copyWith(size: iconSize);
+            })),
+            child: NavigationBar(
+              selectedIndex: currentIndex ?? 0,
+              destinations: destinations,
+              onDestinationSelected: onDestinationSelected,
+            ));
```
Comment (justinmc) [inline packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart:334]:
```suggestion
        return NavigationBarTheme(
          data: currentNavBarTheme.copyWith(
            iconTheme: MaterialStateProperty.resolveWith((Set<MaterialState> states) {
              return currentNavBarTheme.iconTheme
                  ?.resolve(states)
                  ?.copyWith(size: iconSize) ??
                      IconTheme.of(context).copyWith(size: iconSize);
            }),
          ),
          child: NavigationBar(
            selectedIndex: currentIndex ?? 0,
            destinations: destinations,
            onDestinationSelected: onDestinationSelected,
          ),
        );
```

Comment (justinmc) [reply packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart:334]: This formatting was weird compared to the rest of the repo, I think I've corrected it here.

**Review by justinmc — APPROVED** [review]
LGTM with a formatting nit.  Thanks!

**Review by justinmc — APPROVED** [review]
LGTM with a formatting nit.  Thanks!

**Comment by fmt-Println-MKO** [comment]
> LGTM with a formatting nit. Thanks!

now it failed with wrong formatting :(

**Comment by fmt-Println-MKO** [comment]
@justinmc any suggestion on how to solve the formatting issue?
either roll back to the version that worked in the pipeline or any other idea?

**Comment by gspencergoog** [comment]
Just go back to the version that passes.  Formatting is enforced in `flutter/packages`, so there's really only one way to format it that works (what `dartfmt` says).

**Comment by fmt-Println-MKO** [comment]
ok formatting is fixed, it's ready for merge :)

**Pull request merged**